cli/tools/upgrade.rs

```diff
@@ -167,7 +168,25 @@ async fn download_package(
 
   if res.status().is_success() {
     println!("Download has been found");
-    Ok(res.bytes().await?.to_vec())
+
+    let total_size = res.content_length().unwrap() as f64;
+    let mut current_size = 0.0;
+    let mut data = Vec::with_capacity(total_size as usize);
+    let mut stream = res.bytes_stream();
+    while let Some(item) = stream.next().await {
+      let bytes = item?;
+      current_size += bytes.len() as f64;
+      data.extend_from_slice(&bytes);
```

Comment:
1. Print `ESC [1G` to move the cursor to column 1 before printing.
2. Clear line with `ESC [2k` to clear the line without moving the cursor.
3. Print, which should move the cursor to end of line.

Reply:
(Only do step 1 and 2 if stdout is a tty).

Reply:
That wont do, renders the text too slow & laggy, making it extremely difficult to read